com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbePlacement.cs

```diff
@@ -3,236 +3,598 @@
                 }
+
+                brickPositions = new Vector3[maxBrickCountPerAxisInSubCell * maxBrickCountPerAxisInSubCell * maxBrickCountPerAxisInSubCell];
             }
 
-            return num;
+            public void Dispose()
+            {
+                RenderTexture.ReleaseTemporary(sceneSDF);
+                RenderTexture.ReleaseTemporary(sceneSDF2);
+                RenderTexture.ReleaseTemporary(dummyRenderTarget);
+                probeVolumesBuffer.Release();
+
+                for (int i = 0; i <= maxSubdivisionLevelInSubCell; i++)
+                {
+                    bricksBuffers[i].Release();
+                    readbackCountBuffers[i].Release();
+                }
+            }
         }
 
-        static protected int NavPathsToVolumes(ref ProbeReferenceVolume.Volume cellVolume, ref List<ProbeReferenceVolume.Volume> volumes, ref Dictionary<Scene, int> sceneRefs)
+        static readonly int _BricksToClear = Shader.PropertyToID("_BricksToClear");
+        static readonly int _Output = Shader.PropertyToID("_Output");
+        static readonly int _OutputSize = Shader.PropertyToID("_OutputSize");
+        static readonly int _VolumeWorldOffset = Shader.PropertyToID("_VolumeWorldOffset");
+        static readonly int _VolumeSize = Shader.PropertyToID("_VolumeSize");
+        static readonly int _AxisSwizzle = Shader.PropertyToID("_AxisSwizzle");
+        static readonly int _Size = Shader.PropertyToID("_Size");
+        static readonly int _Input = Shader.PropertyToID("_Input");
+        static readonly int _Offset = Shader.PropertyToID("_Offset");
+        static readonly int _ProbeVolumes = Shader.PropertyToID("_ProbeVolumes");
+        static readonly int _ProbeVolumeCount = Shader.PropertyToID("_ProbeVolumeCount");
+        static readonly int _MaxBrickSize = Shader.PropertyToID("_MaxBrickSize");
+        static readonly int _VolumeOffsetInBricks = Shader.PropertyToID("_VolumeOffsetInBricks");
+        static readonly int _Bricks = Shader.PropertyToID("_Bricks");
+        static readonly int _SubdivisionLevel = Shader.PropertyToID("_SubdivisionLevel");
+        static readonly int _MaxSubdivisionLevel = Shader.PropertyToID("_MaxSubdivisionLevel");
+        static readonly int _VolumeSizeInBricks = Shader.PropertyToID("_VolumeSizeInBricks");
+        static readonly int _SDFSize = Shader.PropertyToID("_SDFSize");
+        static readonly int _ProbeVolumeData = Shader.PropertyToID("_ProbeVolumeData");
+        static readonly int _BrickSize = Shader.PropertyToID("_BrickSize");
+        static readonly int _ClearValue = Shader.PropertyToID("_ClearValue");
+
+        static int s_ClearBufferKernel;
+        static int s_ClearKernel;
+        static int s_JumpFloodingKernel;
+        static int s_FillUVKernel;
+        static int s_FinalPassKernel;
+        static int s_VoxelizeProbeVolumesKernel;
+        static int s_SubdivideKernel;
+
+        static ComputeShader _subdivideSceneCS;
```

Comment:
Is this ever released? I don't think it is auto released upon destruction.

Reply:
We don't need to release it (there is no method for it)